modelling/src/main/java/org/axonframework/modelling/command/AnnotationCommandTargetResolver.java

```diff
@@ -70,44 +89,101 @@ public VersionedAggregateIdentifier resolveTarget(CommandMessage<?> command) {
-        } else if (Number.class.isInstance(fieldValue)) {
-            return ((Number) fieldValue).longValue();
-        } else {
-            return Long.parseLong(fieldValue.toString());
-        }
-    }
-}
+	private static Object invokeAnnotated(Message<?> command, Class<? extends Annotation> annotation)
+			throws InvocationTargetException, IllegalAccessException {
+		for (Method m : methodsOf(command.getPayloadType())) {
+			if (AnnotationUtils.isAnnotationPresent(m, annotation)) {
+				ensureAccessible(m);
+				return m.invoke(command.getPayload());
+			}
+		}
+		for (Field f : fieldsOf(command.getPayloadType())) {
+			if (AnnotationUtils.isAnnotationPresent(f, annotation)) {
+				return getFieldValue(f, command.getPayload());
+			}
+		}
+		return null;
+	}
+
+	private Long asLong(Object fieldValue) {
+		if (fieldValue == null) {
+			return null;
+		} else if (Number.class.isInstance(fieldValue)) {
+			return ((Number) fieldValue).longValue();
+		} else {
+			return Long.parseLong(fieldValue.toString());
+		}
+	}
+
+	@Override
+	public String toString() {
+		return "AnnotationCommandTargetResolver [identifierAnnotation=" + identifierAnnotation + ", versionAnnotation="
+				+ versionAnnotation + "]";
+	}
+
+	public static final class Builder {
+		private AnnotationCommandTargetResolver resolver;
+
+		public Builder() {
+			this.resolver = new AnnotationCommandTargetResolver();
+		}
+
+		/**
+		 * Sets the annotation, that marks the target aggregate identifier.
+		 * <p>
+		 * Defaults to {@link TargetAggregateIdentifier}.<br>
+		 * <p>
+		 * If you do not want your messages-module (as inner bounded context "API") to
+		 * be dependent of axon annotations (e.g. to aim empty pom dependencies), then
+		 * you can write your own annotation based on the
+		 * {@link TargetAggregateIdentifier} without referencing the original one (as
+		 * meta-annotation).
+		 * 
+		 * @param annotation - {@link Class} of type {@link Annotation}.
+		 * @return {@link Builder}
+		 */
+		public Builder setTargetAggregateIdentifierAnnotation(Class<? extends Annotation> annotation) {
+			this.resolver.identifierAnnotation = annotation;
+			return this;
+		}
+
+		/**
+		 * Sets the annotation, that marks the target aggregate version.
+		 * <p>
+		 * Defaults to {@link TargetAggregateVersion}.
+		 * <p>
+		 * If you do not want your messages-module (as inner bounded context "API") to
+		 * be dependent of axon annotations (e.g. to aim empty pom dependencies), then
+		 * you can write your own annotation based on the {@link TargetAggregateVersion}
+		 * without referencing the original one (as meta-annotation).
+		 * 
+		 * @param annotation - {@link Class} of type {@link Annotation}.
+		 * @return {@link Builder}
+		 */
+		public Builder setTargetAggregateVersionAnnotation(Class<? extends Annotation> annotation) {
+			this.resolver.versionAnnotation = annotation;
+			return this;
+		}
+
+		public AnnotationCommandTargetResolver build() {
+			try {
```

Comment:
I think it's nicer to call the `AnnotationCommandTargetResolver` constructor here, with, as suggested earlier, `this` as the sole parameter, so that the `AnnotationCommandTargetResolver` constructor can pull the `identifierAnnotation` and `versionAnnotation` from it.